### PR TITLE
Add react docgen info into React classes with the react-docgen babel plugin

### PR DESCRIPTION
**Pull request by madushan1000**
Generates propType docs for all react components used in stories at runtime using [babel-plugin-react-docgen](https://github.com/kadirahq/babel-plugin-react-docgen). 
It also collects all the react propType docs to a global object named `STORYBOOK_REACT_CLASSES`.

### CODE REVIEW

File: src/server/config/docgen.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/* globals window */
```
Comment (arunoda) [inline src/server/config/docgen.js]: Change this filename as `globals.js`

**Comment by jeffijoe** [comment]
Is it possible to disable this? It's causing some of my stuff to break.

**Comment by arunoda** [comment]
Could I know the problem you got? More info on the error?
And yes we should provide a way to stop it.

**Comment by jeffijoe** [comment]
It was a reference error, `Sink is not defined` where Sink was my component. I made a HoC of sorts that returns 2 components and it could apparently not deal with it.

**Comment by arunoda** [comment]
@jeffijoe can I have a sample version of that class? We'll work on a way to turn this off or something similar.

Could you create an new issue with info?

**Comment by jeffijoe** [comment]
@arunoda done, #580

**Pull request merged**